src/jit/lsrabuild.cpp

```diff
@@ -1327,9 +1327,8 @@ void LinearScan::buildInternalRegisterUses()
 VARSET_VALRET_TP
 LinearScan::buildUpperVectorSaveRefPositions(GenTree* tree, LsraLocation currentLoc, regMaskTP fpCalleeKillSet)
 {
-    assert(enregisterLocalVars);
     VARSET_TP liveLargeVectors(VarSetOps::MakeEmpty(compiler));
-    if (!VarSetOps::IsEmpty(compiler, largeVectorVars))
+    if (enregisterLocalVars && !VarSetOps::IsEmpty(compiler, largeVectorVars))
```

Comment:
Could you please clarify some names, overall we have `variables` and `temps`, `largeVectorVars` refers to `variables`, `enregisterLocalVars` also refers to variables, is it correct? 
Could we have non-empty `largeVectorVars` (that doesn't include temps, only vars) when `enregisterLocalVars ` is `false`? Sound like we can't, then can you move the deleted assert under this condition?

Reply:
Looks like this function does two things that needs to be split:
1. a function that returns the set of lclVars (excluding temps) that are killed by this node;
2. a function that create special RefPositions for saving the upper half of a set of large vector (including temps).

and the same for `buildUpperVectorRestoreRefPositions`.

If you do not want to split them now, could you please update their headers to include that they not only create re positions but also save/restore vars.

Reply:
It would make sense to split out the determination of the live large vector lclVars (including lclVar temps, which are the same as lclVars from the perspective of LSRA) from the building of the RefPositions). I'd prefer to do that as a separate refactor.
Note, though, that this method doesn't actually save or restore anything, it just creates the RefPositions that *may* cause their upper half to be saved & restored, at allocation time, *if* they reside in a callee-save register.

Reply:
> the determination of the live large vector lclVars (including lclVar temps, which are the same as lclVars from the perspective of LSRA)

Do you mean `liveLargeVectors` that the method returns? We calculate `liveLargeVectors` only under `if (enregisterLocalVars && !VarSetOps::IsEmpty(compiler, largeVectorVars))` as I see.  How can it include lclVar temp when we iterate only through vars? Could you please rename `liveLargeVectors` to `liveLargeVectorsVars` here as well?

Note that the return value is calculated in the first part of this function and the second part does not affect it at all.